lib/state/simperium/middleware.ts

```diff
@@ -246,27 +246,45 @@ export const initSimperium = (
       case 'FILTER_NOTES':
       case 'OPEN_NOTE':
       case 'SELECT_NOTE': {
+        const state = getState();
         const noteId =
           action.noteId ??
           action.meta?.nextNoteToOpen ??
           getState().ui.openedNote;
 
-        if (noteId) {
+        //  Preload the revisions when opening a note but only do it if no revisions are in memory
+        if (noteId && !state.data.noteRevisions.get(noteId)?.size) {
```

Comment:
let's leave `getState().ui.openedNote === noteId` in there. that is actually a guard against this same kind of proliferation. what's happening is that when jumping through the list of notes we end up opening every one for about 50ms and we don't want to issue requests for all those all at once.